metric.go

```diff
@@ -14,18 +14,14 @@ var ErrInvalidOrgIdzero = errors.New("org-id cannot be 0")
 var ErrInvalidEmptyName = errors.New("name cannot be empty")
 var ErrInvalidMtype = errors.New("invalid mtype")
 var ErrInvalidTagFormat = errors.New("invalid tag format")
+var ErrUnknownPartitionMethod = errors.New("unknown partition method")
 
 type PartitionedMetric interface {
 	Validate() error
 	SetId()
-	// return a []byte key comprised of the metric's OrgId
-	// accepts an input []byte to allow callers to re-use
-	// buffers to reduce memory allocations
-	KeyByOrgId([]byte) []byte
-	// return a []byte key comprised of the metric's Name
-	// accepts an input []byte to allow callers to re-use
-	// buffers to reduce memory allocations
-	KeyBySeries([]byte) []byte
+	// using the provided partitionByMethod, and number of partitions being used
+	// generate and return the partition id that should be used with this metric.
+	PartitionID(method PartitionByMethod, partitions int32) (int32, error)
```

Comment:
Are there any other projects aside from Metrictank and tsdb-gw that use this interface? I suppose we will need to update them all at the same time. I know there are some PRs open to deal with it, but I don't think the MT PR is ready yet #grafana/metrictank/pull/1282

Reply:
I'm only aware of TSDB and MT (in multiple locations). Once this is merged, MT and TSDB can update their vendored schema and start using the new partitioning methods, but there's nothing stopping us from merging this PR already before MT & TSDB integrated it.

Reply:
> there's nothing stopping us from merging this PR already before MT & TSDB integrated it.

in fact that's why the mentioned MT pr is stale, because it's waiting for this one to be done so it can pull this code in cleanly.